### ISSUE TITLE
test twisted.web.domhelpers against xml.dom.minidom

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3600]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3600|
|Type|enhancement|
|Created|2009-01-05 18:22:04Z|
|Branch|https://github.com/twisted/twisted/tree/minidom-helpers-3600|

`twisted.web.domhelpers` uses what are basically generic, standard DOM APIs to do what it does.  So it should be able to work against more than one DOM implementation.  The world is terrible though, so it doesn't actually.  It should be easy to run all of its tests against both minidom and microdom, though, and make adjustments where it doesn't currently work with minidom.

This is a step towards [#3560](https://github.com/twisted/twisted/issues/3560).

<details><summary>Searchable metadata</summary>

```
trac-id__3600 3600
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_minidom_helpers_3600 branches/minidom-helpers-3600
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1231179724000000 1231179724000000
changetime__1231515832000000 1231515832000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [25791]) Branching to 'minidom-helpers-3600'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Okey dokey

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Can you remove the remaining calls to 'assert' in test_domhelpers? Once done, please merge.
Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Fixed in r25916

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [25917]) Merge minidom-helpers-3600

Author: exarkun
Reviewer: therve
Fixes: [#3600](https://github.com/twisted/twisted/issues/3600)

Change the existing `twisted.web.domhelpers` test suite so that it also runs
using `xml.dom.minidom` as the DOM implementation, in addition to still running
while using `twisted.web.microdom` as the DOM implementation.  Make the necessary
changes to `twisted.web.domhelpers` so that the suite passes with both implementations.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
